src/test/regress/sql/create_role_propagation.sql

```diff
@@ -44,6 +46,8 @@ CREATE ROLE "create_role""edge";
 -- test grant role
 GRANT create_group TO create_role;
 GRANT create_group TO create_role_2 WITH ADMIN OPTION;
+GRANT create_group TO create_role_3 WITH INHERIT;
```

Comment:
These tests are good, but there is a small issue with the Postgres version. In Citus, we keep PG version compatible tests for the latest 3 major versions. `INHERIT` and `SET` commands with fail with PG14-PG15.
To avoid this, we have already created a test file specific for PG16. See `pg16.sql` and add the GRANT command tests there.